controller/invitation.go

```diff
@@ -16,47 +15,47 @@ import (
 // InvitationController implements the invitation resource.
 type InvitationController struct {
 	*goa.Controller
-	db         application.DB
-	invService invitationService.InvitationService
+	db application.DB
 }
 
 // NewInvitationController creates a invitation controller.
-func NewInvitationController(service *goa.Service, db application.DB, invitationService invitationService.InvitationService) *InvitationController {
-	return &InvitationController{Controller: service.NewController("InvitationController"), db: db, invService: invitationService}
+func NewInvitationController(service *goa.Service, db application.DB) *InvitationController {
+	return &InvitationController{Controller: service.NewController("InvitationController"), db: db}
 }
 
 // Create runs the create action.
 func (c *InvitationController) CreateInvite(ctx *app.CreateInviteInvitationContext) error {
 	currentIdentity, err := login.LoadContextIdentityIfNotDeprovisioned(ctx, c.db)
 	if err != nil {
-		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError(err.Error()))
+		return jsonapi.JSONErrorResponse(ctx, err)
 	}
 
 	var invitations []invitation.Invitation
 
 	for _, invitee := range ctx.Payload.Data {
 		// Validate that an identifying parameter has been set
-		if invitee.IdentityID == nil && invitee.Username == nil && invitee.UserEmail == nil {
+		if invitee.IdentityID == nil {
 			return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterErrorFromString("user identifier", "", "no identifier provided"))
 		}
 
 		// If an identity ID has been provided for the user, convert it to a UUID here
-		var identityID *uuid.UUID
+		var identityID uuid.UUID
 		if invitee.IdentityID != nil && *invitee.IdentityID != "" {
-			*identityID, err = uuid.FromString(*invitee.IdentityID)
+			identityID, err = uuid.FromString(*invitee.IdentityID)
 		}
 
 		// Create the Invitation object, and append it to our list of invitations
 		invitations = append(invitations, invitation.Invitation{
-			IdentityID: identityID,
-			UserName:   invitee.Username,
-			UserEmail:  invitee.UserEmail,
+			IdentityID: &identityID,
```

Comment:
IdentityID here is actually required, right? So, we can make it UUID (not *UUID).